iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -489,10 +489,29 @@ void ProcessManager::addPublisherForProcess(const RuntimeName_t& name,
             {
                 runtime::IpcMessage sendBuffer;
                 sendBuffer << runtime::IpcMessageTypeToString(runtime::IpcMessageType::ERROR);
-                sendBuffer << runtime::IpcMessageErrorTypeToString( // map error codes
-                    (maybePublisher.get_error() == PortPoolError::UNIQUE_PUBLISHER_PORT_ALREADY_EXISTS
-                         ? runtime::IpcMessageErrorType::NO_UNIQUE_CREATED
-                         : runtime::IpcMessageErrorType::PUBLISHER_LIST_FULL));
+
+                std::string error;
+                switch (maybePublisher.get_error())
+                {
+                case PortPoolError::UNIQUE_PUBLISHER_PORT_ALREADY_EXISTS:
+                {
+                    error = runtime::IpcMessageErrorTypeToString(runtime::IpcMessageErrorType::NO_UNIQUE_CREATED);
+                    break;
+                }
+                case PortPoolError::INTERNAL_SERVICE_DESCRIPTION_IS_FORBIDDEN:
+                {
+                    error = runtime::IpcMessageErrorTypeToString(
+                        runtime::IpcMessageErrorType::SERVICE_DESCRIPTION_IS_FORBIDDEN);
```

Comment:
Why do we not want to propagate the reason the service description is not allowed? (The error is more generic than it needs to be IMO).

Do we have other scenarios of service descriptions being not allowed (I am not aware of them at least).

Reply:
There could be reasons in future, but YAGNI. I adapted the error name.